pkg/networkmanager/firewall.jsx

```diff
@@ -471,39 +481,19 @@ class AddServicesModal extends React.Component {
         else
             services = this.state.services;
 
-        // hide services which have been enabled in all zones
-        if (services) {
-            services = services.filter(s => {
-                let allZonesContainService = true;
-                for (const zone of firewall.activeZones)
-                    allZonesContainService &= firewall.zones[zone].services.indexOf(s.id) !== -1;
-                return !allZonesContainService;
-            });
-        }
+        // hide services which have been enabled in the zone
+        if (services)
+            services = services.filter(s => firewall.zones[this.props.zoneId].services.indexOf(s.id) === -1);
 
-        var addText = this.state.custom ? _("Add Ports") : _("Add Services");
-        var zonesText = this.state.custom ? _("Add ports to the following zones:") : _("Add services to following zones:");
+        var addText = cockpit.format(this.state.custom ? _("Add Ports to $0 zone") : _("Add Services to $0 zone"),
```

Comment:
This doesn't seem to scale very well.

![Screenshot from 2019-09-26 18-52-01](https://user-images.githubusercontent.com/209895/65708382-bf3dff80-e08e-11e9-91d8-1947d544bad1.png)

I would suggest making the button only say `Add Services`, without any mention of the zone. It's already in the dialog header.